crates/rooch-key/src/keystore.rs

```diff
@@ -38,237 +30,153 @@ use std::fs::File;
 use std::io::BufReader;
 use std::path::{Path, PathBuf};
 
-pub struct ImportedMnemonic<Addr> {
-    pub address: Addr,
-    pub encryption: EncryptionData,
-}
-
-pub struct UpdatedAddress<KeyPair> {
-    pub key_pair: KeyPair,
+pub struct ImportedMnemonic {
+    pub address: RoochAddress,
     pub encryption: EncryptionData,
 }
 
 #[derive(Serialize, Deserialize, Debug)]
 #[enum_dispatch(AccountKeystore)]
-pub enum Keystore<K: Ord> {
-    File(FileBasedKeystore<K>),
-    InMem(InMemKeystore<K>),
+pub enum Keystore {
+    File(FileBasedKeystore),
+    InMem(InMemKeystore),
 }
 
-#[enum_dispatch]
-pub trait AccountKeystore<Addr: Copy, PubKey, KeyPair, Sig, TransactionData>: Send + Sync {
-    type Transaction;
-
-    fn add_encryption_data_by_key_pair_type(
+pub trait AccountKeystore {
+    fn add_encryption_data(
         &mut self,
-        address: Addr,
-        key_pair_type: KeyPairType,
+        address: RoochAddress,
         encryption: EncryptionData,
     ) -> Result<(), anyhow::Error>;
     fn get_address_public_keys(
         &self,
         password: Option<String>,
-    ) -> Result<Vec<(Addr, PubKey)>, RoochError>;
-    fn get_public_key_by_key_pair_type(
-        &self,
-        key_pair_type: KeyPairType,
-        password: Option<String>,
-    ) -> Result<PubKey, anyhow::Error>;
+    ) -> Result<Vec<(RoochAddress, PublicKey)>, RoochError>;
+    fn get_public_key(&self, password: Option<String>) -> Result<PublicKey, anyhow::Error>;
     fn get_key_pairs(
         &self,
-        address: &Addr,
+        address: &RoochAddress,
         password: Option<String>,
-    ) -> Result<Vec<KeyPair>, anyhow::Error>;
-    fn get_key_pair_by_type_password(
+    ) -> Result<Vec<RoochKeyPair>, anyhow::Error>;
+    fn get_key_pair_by_password(
```

Comment:
Would `get_key_pair_by_password ` -> `get_key_pair_with_password ` be more appropriate?

Reply:
Yep. It's just a naming.